ext/simplexml/tests/current_error.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+SimpleXML: invoking current() after the iterator has already been consumed
+--EXTENSIONS--
+simplexml
+--FILE--
+<?php
+
+$xml =<<<EOF
+<?xml version='1.0'?>
+<root>
+<elem/>
+</root>
+EOF;
+
+$sxe = simplexml_load_string($xml);
+
+try {
+    $sxe->current();
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+foreach($sxe as $name => $data) {
+    var_dump($name);
+}
```

Comment:
It would be better to make this something like
```
for ($sxe->rewind(); $sxe->valid(); $sxe->next()) {
    var_dump($sxe->key(), $sxe->current());
}
```
because foreach goes through a different implementation.

Reply:
Fixed while applying